solr/contrib/ltr/src/java/org/apache/solr/ltr/ranking/RankSVMModel.java

```diff
@@ -28,36 +29,62 @@
 
 public class RankSVMModel extends LTRScoringModel {
 
-  protected float[] featureToWeight;
+  protected Float[] featureToWeight;
 
   /** name of the attribute containing the weight of the SVM model **/
   public static final String WEIGHTS_PARAM = "weights";
 
-  public RankSVMModel(String name, List<Feature> features,
+  public static RankSVMModel create(String name, List<Feature> features,
```

Comment:
Good point, yes, that's the plan. `RankSVMModel.create` here will have a short life and will be superseded by a `getInstance()` function in a later pull request (this is only first step towards plugin style models). Also note that this pull request did therefore not add a `create` method to the other two model classes (because none of the existing tests needed it).